make/tools.mk

```diff
@@ -16,36 +17,38 @@
 ##############################
 
 # Set up ARM (STM32) SDK
-ARM_SDK_DIR ?= $(TOOLS_DIR)/gcc-arm-none-eabi-10.3-2021.10
+ARM_SDK_DIR ?= $(TOOLS_DIR)/arm-gnu-toolchain-12.2.mpacbti-rel1
 # Checked below, Should match the output of $(shell arm-none-eabi-gcc -dumpversion)
-GCC_REQUIRED_VERSION ?= 10.3.1
-
-.PHONY: arm_sdk_version
-
-arm_sdk_version:
-	$(V1) $(ARM_SDK_PREFIX)gcc --version
+GCC_REQUIRED_VERSION ?= 12.2.1
 
 ## arm_sdk_install   : Install Arm SDK
 .PHONY: arm_sdk_install
 
-ARM_SDK_URL_BASE  := https://developer.arm.com/-/media/Files/downloads/gnu-rm/10.3-2021.10/gcc-arm-none-eabi-10.3-2021.10
-# source: https://developer.arm.com/open-source/gnu-toolchain/gnu-rm/downloads
+# source: https://developer.arm.com/downloads/-/arm-gnu-toolchain-downloads
 ifeq ($(OSFAMILY), linux)
-  ARM_SDK_URL  := $(ARM_SDK_URL_BASE)-x86_64-linux.tar.bz2
+  ARM_SDK_URL := https://developer.arm.com/-/media/Files/downloads/gnu/12.2.mpacbti-rel1/binrel/arm-gnu-toolchain-12.2.mpacbti-rel1-x86_64-arm-none-eabi.tar.xz?rev=71e595a1f2b6457bab9242bc4a40db90&hash=37B0C59767BAE297AEB8967E7C54705BAE9A4B95
+  ARM_SDK_DIR := $(ARM_SDK_DIR)-x86_64-arm-none-eabi
 endif
 
 ifeq ($(OSFAMILY), macosx)
-  ARM_SDK_URL  := $(ARM_SDK_URL_BASE)-mac.tar.bz2
+  ARM_SDK_URL := https://developer.arm.com/-/media/Files/downloads/gnu/12.2.mpacbti-rel1/binrel/arm-gnu-toolchain-12.2.mpacbti-rel1-darwin-x86_64-arm-none-eabi.tar.xz?rev=c517551be7e3468d826036530848cde0&hash=25A1A3E625D6822B6F7E1705D11B623E07EDEC3F
+  ARM_SDK_DIR := $(ARM_SDK_DIR)-darwin-x86_64-arm-none-eabi
 endif
 
 ifeq ($(OSFAMILY), windows)
-  ARM_SDK_URL  := $(ARM_SDK_URL_BASE)-win32.zip
+  ARM_SDK_URL := https://developer.arm.com/-/media/Files/downloads/gnu/12.2.mpacbti-rel1/binrel/arm-gnu-toolchain-12.2.mpacbti-rel1-mingw-w64-i686-arm-none-eabi.zip?rev=3a67072e4b984fb2911b4953c0fa6f53&hash=FD03ED6AF3D23D37E047708C49F9E16D17A85311
+  ARM_SDK_DIR := $(ARM_SDK_DIR)-mingw-w64-i686-arm-none-eabi
 endif
 
 ARM_SDK_FILE := $(notdir $(ARM_SDK_URL))
```

Comment:
Does this strip `?rev=...`  part correctly?

Reply:
Yes: 

tools/arm-gnu-toolchain-12.2.mpacbti-rel1-x86_64-arm-none-eabi